Mathlib/Data/Finset/Pi.lean

```diff
@@ -139,6 +140,18 @@ theorem pi_disjoint_of_disjoint {δ : α → Type*} {s : Finset α} (t₁ t₂ :
       congr_fun (congr_fun eq₁₂ a) ha
 #align finset.pi_disjoint_of_disjoint Finset.pi_disjoint_of_disjoint
 
-end Pi
+/-! ### Diagonal -/
+
+variable {ι : Type*} [DecidableEq (ι → α)] {s : Finset α} {f : ι → α}
+
+/-- The diagonal of a finset `s : Finset α` as a finset of functions `ι → α`, namely the set of
+constant functions valued in `s`. -/
+def piDiag (s : Finset α) (ι : Type*) [DecidableEq (ι → α)] : Finset (ι → α) := s.image (const ι)
```

Comment:
Would `constFnsWithin` be better? I think a niche notion like this can afford to have a long name, and should have a name that is clear.

Reply:
I see your point, but it is completely analogous to `Finset.diag`, except for `ι → α` rather than `α × α`. I think this justifies the current name (and points to the fact that we might want the `ι →₀ α` version in the future